lib/contextual_logger/redactor.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module ContextualLogger
+  class Redactor
+    attr_reader :redaction_set, :redaction_regex
+
+    def initialize
+      @redaction_set   = Set.new
+      @redaction_regex = nil
+    end
+
+    def match(sensitive_data)
+      if redaction_set.add?(Regexp.escape(sensitive_data))
+        @redaction_regex = Regexp.new(
+          redaction_set.to_a.join('|')
```

Comment:
If we ever support callers setting their own Regexps (I think they'll want that) this will break down. Probably simpler just to do `.any?` across a list.

Reply:
Regex support _could_ come later.  YAGIN

Reply:
But worth doing a Regexp escape and join here vs simply
```
redaction_set.any? { |redaction_pattern| sensitive_data.include?(redaction_pattern) }
```